File: apps/web/src/lib/media.ts
import { apiPost } from "./api";

export type SasResponse = { uploadUrl: string; blobUrl: string; expiresAt: string };

export async function requestUploadUrl(args: {
  type: "photo" | "audio";
  contentType: string;
  fileName: string;
}): Promise<SasResponse> {
  return apiPost<SasResponse>("/media/sas", args);
}

export async function completeUpload(args: {
  blobUrl: string;
  type: "photo" | "audio";
  metadata?: Record<string, unknown>;
}): Promise<{ assetId: string }> {
  return apiPost<{ assetId: string }>("/media/complete", args);
}

File: services/api/app/schemas/media.py
from pydantic import BaseModel


class SasRequest(BaseModel):
    type: str  # "photo" | "audio"
    contentType: str
    fileName: str


class SasResponse(BaseModel):
    uploadUrl: str
    blobUrl: str
    expiresAt: str


class CompleteRequest(BaseModel):
    blobUrl: str
    type: str  # "photo" | "audio"
    metadata: dict | None = None


class CompleteResponse(BaseModel):
    assetId: str
